contracts/blockchain_parameters/blockchain_parameters.go

```diff
@@ -0,0 +1,209 @@
+		  }
+		],
+		"payable": false,
+		"stateMutability": "view",
+		"type": "function"
+	},
+	{
+		"constant": true,
+		"inputs": [],
+		"name": "getUptimeLookbackWindow",
+		"outputs": [
+			{
+				"internalType": "uint256",
+				"name": "lookbackWindow",
+				"type": "uint256"
+			}
+		],
+		"payable": false,
+		"stateMutability": "view",
+		"type": "function"
+	},
+	{
+		"constant": true,
+		"inputs": [],
+		"name": "intrinsicGasForAlternativeFeeCurrency",
+		"outputs": [
+			{
+				"name": "",
+				"type": "uint256"
+			}
+		],
+		"payable": false,
+		"stateMutability": "view",
+		"type": "function"
+	}
+]`
+)
+
+var (
+	getMinimumClientVersionMethod               *contracts.BoundMethod
+	intrinsicGasForAlternativeFeeCurrencyMethod *contracts.BoundMethod
+	blockGasLimitMethod                         *contracts.BoundMethod
+	getUptimeLookbackWindowMethod               *contracts.BoundMethod
+)
+
+func init() {
+	parsedAbi, err := abi.JSON(strings.NewReader(blockchainParametersABIString))
+	if err != nil {
+		log.Crit("Error reading ABI for BlockchainParameters", "err", err)
+	}
+
+	getMinimumClientVersionMethod = contracts.NewRegisteredContractMethod(params.BlockchainParametersRegistryId, &parsedAbi, "getMinimumClientVersion", params.MaxGasForReadBlockchainParameter)
+	intrinsicGasForAlternativeFeeCurrencyMethod = contracts.NewRegisteredContractMethod(params.BlockchainParametersRegistryId, &parsedAbi, "intrinsicGasForAlternativeFeeCurrency", params.MaxGasForReadBlockchainParameter)
+	blockGasLimitMethod = contracts.NewRegisteredContractMethod(params.BlockchainParametersRegistryId, &parsedAbi, "blockGasLimit", params.MaxGasForReadBlockchainParameter)
+	getUptimeLookbackWindowMethod = contracts.NewRegisteredContractMethod(params.BlockchainParametersRegistryId, &parsedAbi, "getUptimeLookbackWindow", params.MaxGasForReadBlockchainParameter)
+}
+
+func GetMinimumVersion(vmRunner vm.EVMRunner) (*params.VersionInfo, error) {
+	version := [3]*big.Int{big.NewInt(0), big.NewInt(0), big.NewInt(0)}
+	err := getMinimumClientVersionMethod.Query(vmRunner, &version)
+	if err != nil {
+		return nil, err
+	}
+	return &params.VersionInfo{
+		Major: version[0].Uint64(),
+		Minor: version[1].Uint64(),
+		Patch: version[2].Uint64(),
+	}, nil
+}
+
+func GetIntrinsicGasForAlternativeFeeCurrency(vmRunner vm.EVMRunner) uint64 {
+	var gas *big.Int
+	err := intrinsicGasForAlternativeFeeCurrencyMethod.Query(vmRunner, &gas)
+
+	if err != nil {
+		log.Trace("Default gas", "gas", params.IntrinsicGasForAlternativeFeeCurrency, "method", "intrinsicGasForAlternativeFeeCurrency")
+		return params.IntrinsicGasForAlternativeFeeCurrency
+	}
+	log.Trace("Reading gas", "gas", gas)
+	return gas.Uint64()
+}
+
+func GetBlockGasLimitOrDefault(vmRunner vm.EVMRunner) uint64 {
+	val, err := GetBlockGasLimit(vmRunner)
+	if err != nil {
+		logError("blockGasLimit", err)
+		return params.DefaultGasLimit
+	}
+	return val
+}
+
+func GetBlockGasLimit(vmRunner vm.EVMRunner) (uint64, error) {
```

Comment:
Is this function intended to possibly be used directly anywhere?  If not, I would make it private or even inline it like it originally was (and which would make it be more similar to `GetIntrinsicGasForAlternativeFeeCurrency()` which handles the error case inline).

Reply:
The intent here was to make explicit that there's an error and we are using a default. That's why there's `GetBlockGasLimitOrDefault`. I think most of the time we are not really aware of using defaults, and we should be. 

I'm not sure yet, but i was thinking maybe defaults should live elsewhere, more on the caller side, and not on the fn implementation side. So this module would have `GetBlockGasLimit` and the caller will be the one handling the default, and thus be more explicit about defaults

Reply:
I kind of like the default being together with the contract call (together as in in the same file).  But it does make sense to have separate functions to make that clearer.  If so, I would (a) make the other function private, (b) make the other methods also be the same way.

Reply:
I did (b), and added godocs comments.

Not sure if you were suggesting `a & b` or `a | b`. I think having them public is not bad. True, they are not being used today, but doesn't seem wrong from a "what's my public interface" perspective as a module designer.

Reply:
I was suggesting (a) and (b).  My inclination is to make it private unless we have a use-case or envision a future use-case.  It helps prevent accidentally using the wrong one without realizing it's the wrong one.  And it's always possible to make it public if we have a reason to in the future, and if we do that then it's having been private until now won't cause any problems.